Mathlib/FieldTheory/SeparableClosure.lean

```diff
@@ -0,0 +1,386 @@
+`separableClosure F K` is equal to `separableClosure E K`. -/
+theorem separableClosure.eq_restrictScalars_of_isSeparable [Algebra E K] [IsScalarTower F E K]
+    [IsSeparable F E] : separableClosure F K = (separableClosure E K).restrictScalars F := by
+  refine le_antisymm (separableClosure.le_restrictScalars F E K) fun x hx ↦ ?_
+  rw [mem_restrictScalars, mem_separableClosure_iff,
+    ← isSeparable_adjoin_simple_iff_separable] at hx
+  haveI : IsSeparable F (E⟮x⟯.restrictScalars F) := IsSeparable.trans F E E⟮x⟯
+  have h : x ∈ E⟮x⟯.restrictScalars F := mem_adjoin_simple_self E x
+  exact (mem_separableClosure_iff F _).2 <| separable_of_mem_isSeparable F _ h
+
+/-- If `K / E / F` is a field extension tower, then `E` adjoin `separableClosure F K` is contained
+in `separableClosure E K`. -/
+theorem separableClosure.adjoin_le [Algebra E K] [IsScalarTower F E K] :
+    adjoin E (separableClosure F K) ≤ separableClosure E K :=
+  adjoin_le_iff.2 <| le_restrictScalars F E K
+
+/-- A compositum of two separable extensions is separable. -/
+instance IntermediateField.isSeparable_sup (L1 L2 : IntermediateField F E)
+    [h1 : IsSeparable F L1] [h2 : IsSeparable F L2] :
+    IsSeparable F (L1 ⊔ L2 : IntermediateField F E) := by
+  rw [← le_separableClosure_iff] at h1 h2 ⊢
+  exact sup_le h1 h2
+
+/-- A compositum of separable extensions is separable. -/
+instance IntermediateField.isSeparable_iSup {ι : Type*} {t : ι → IntermediateField F E}
+    [h : ∀ i, IsSeparable F (t i)] : IsSeparable F (⨆ i, t i : IntermediateField F E) := by
+  simp_rw [← le_separableClosure_iff] at h ⊢
+  exact iSup_le h
+
+end separableClosure
+
+namespace Field
+
+/-- The (infinite) separable degree for a general field extension `E / F` is defined
+to be the degree of `(separableClosure F E) / F`. -/
+def sepDegree := Module.rank F (separableClosure F E)
+
+/-- The (infinite) inseparable degree for a general field extension `E / F` is defined
+to be the degree of `E / (separableClosure F E)`. -/
+def insepDegree := Module.rank (separableClosure F E) E
+
+/-- The (finite) inseparable degree for a general field extension `E / F` is defined
+to be the degree of `E / (separableClosure F E)` as a natural number. It is defined to be zero
+if such field extension is infinite. -/
+def finInsepDegree : ℕ := Cardinal.toNat (insepDegree F E)
+
+instance instNeZeroSepDegree : NeZero (sepDegree F E) := ⟨rank_pos.ne'⟩
+
+instance instNeZeroInsepDegree : NeZero (insepDegree F E) := ⟨rank_pos.ne'⟩
+
+instance instNeZeroFinInsepDegree [FiniteDimensional F E] :
+    NeZero (finInsepDegree F E) := ⟨finrank_pos.ne'⟩
+
+/-- If `E` and `K` are isomorphic as `F`-algebras, then they have the same (infinite)
+separable degree over `F`. -/
+theorem sepDegree_eq_of_equiv (i : E ≃ₐ[F] K) :
+    Cardinal.lift.{w} (sepDegree F E) = Cardinal.lift.{v} (sepDegree F K) :=
+  (separableClosure.algEquivOfAlgEquiv F E K i).toLinearEquiv.lift_rank_eq
```

Comment:
Can you provide a version of this without `lift`?
Then `sepDegree_bot` won't need those rewrites for example.

Reply:
Sure. But `sepDegree_bot` still needs the rewrites since `F` and `⊥` are not in the same universe 😂